src/main/java/toughasnails/handler/thirst/FillBottleHandler.java

```diff
@@ -38,14 +40,15 @@ public void onPlayerRightClickWater(PlayerInteractEvent.RightClickItem event) th
         ItemStack stack = player.getHeldItem(event.getHand());
         World world = player.worldObj;
 
-        if (stack.getItem().equals(Items.GLASS_BOTTLE) && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
+        if (stack != null && stack.getItem().equals(Items.GLASS_BOTTLE) && SyncedConfig.getBooleanValue(GameplayOption.ENABLE_THIRST))
         {
-
-
-            EnumActionResult actionResult = stack.getItem().onItemRightClick(new ItemStack(stack.getItem(), 0), event.getWorld(), player, event.getHand()).getType();
-
-            // Only substitute water bottles with dirty water bottles
-            if (actionResult.equals(EnumActionResult.SUCCESS))
+        	
+            ActionResult actionResult = stack.getItem().onItemRightClick(new ItemStack(stack.getItem(), 0), event.getWorld(), player, event.getHand());
+			ItemStack resultStack = ((ItemStack)actionResult.getResult());
```

Comment:
Hrm, are you using Linux by any chance? I think both Windows and OSX have issues with it